False positive in `test_case_accessibility` for variables used in `init`

### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

When defining variables with `let` or `var` inside the `init` block, `test_case_accessibility` warning is raised, when it shouldn't be. Maybe they're falsely being treated as properties?

##### Complete output when running SwiftLint, including the stack trace and command used

Command output:
```bash
ProjectTests % swiftlint BaseTestCase.swift
Linting Swift files at paths BaseTestCase.swift
Linting 'BaseTestCase.swift' (1/1)
.../Project/Tests/ProjectTests/BaseTestCase.swift:6:9: warning: Test Case Accessibility Violation: Test cases should only contain private non-test members (test_case_accessibility)
.../Project/Tests/ProjectTests/BaseTestCase.swift:7:9: warning: Test Case Accessibility Violation: Test cases should only contain private non-test members (test_case_accessibility)
Done linting! Found 2 violations, 0 serious in 1 file.
```

### Environment

* SwiftLint version: 0.52.4
* Installation method used: Homebrew
* Paste your configuration file:

```yml
opt_in_rules:
  - test_case_accessibility
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations): No
* Which Xcode version are you using: Xcode 14.3.1 Build version 14E300c
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
import XCTest

class BaseTestCase: XCTestCase {
    override init() {
        super.init()
        let foo = 1 // test_case_accessibility
        var bar = 2 // test_case_accessibility
    }
}
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -153,6 +153,10 @@
   [Martin Redington](https://github.com/mildm8nnered)
   [#4798](https://github.com/realm/SwiftLint/issues/4798)
 
+* Fix false positive in the `test_case_accessibility` rule.  
+  [gibachan](https://github.com/gibachan)
+  [#5211](https://github.com/realm/SwiftLint/issues/5211)
+
 ## 0.52.4: Lid Switch
 
 #### Breaking

--- a/Source/SwiftLintBuiltInRules/Rules/Lint/TestCaseAccessibilityRuleExamples.swift
+++ b/Source/SwiftLintBuiltInRules/Rules/Lint/TestCaseAccessibilityRuleExamples.swift
@@ -50,6 +50,16 @@ internal struct TestCaseAccessibilityRuleExamples {
 
         Example("""
         class FooTests: XCTestCase {
+            override init() {
+                super.init()
+                let foo = 1
+                var bar = 2
+            }
+        }
+        """),
+
+        Example("""
+        class FooTests: XCTestCase {
             func allowedPrefixTestFoo() {}
         }
         """, configuration: ["allowed_prefixes": ["allowedPrefix"]]),

--- a/Source/SwiftLintCore/Visitors/ViolationsSyntaxVisitor.swift
+++ b/Source/SwiftLintCore/Visitors/ViolationsSyntaxVisitor.swift
@@ -82,6 +82,7 @@ public extension Array where Element == DeclSyntaxProtocol.Type {
         EnumDeclSyntax.self,
         FunctionDeclSyntax.self,
         ExtensionDeclSyntax.self,
+        InitializerDeclSyntax.self,
         ProtocolDeclSyntax.self,
         StructDeclSyntax.self,
         VariableDeclSyntax.self

